### DISCUSSION TITLE
ValueError: truncated dump file; tried to read 4 file header bytes, only got 0

### DISCUSSION THREAD

**Discussion by DarmJarlo** [nfstream/nfstream#160] [Q&A]
ValueError: truncated dump file; tried to read 4 file header bytes, only got 0

Hey ,I'm trying to convert collected packets( in a pcap file) to flows through NFstream everytime when I collected 7 packets. And then I try to convert this stream to a csv file. Unfortunately it always gave me a ValueError like above. It seems that the stream hasn't been successfully produced when I read it. I tried to write a while loop to reproduce the stream but it never worked( just trapped in this loop) .Could someone help to fix this problem? Thanks!

```
 while(not os.path.exists(self.csv_path)):
                try:
                    my_stream = self.create_mystream(self.address_pair_pcapfile[addr_pair])
                    my_stream.to_csv(path=self.csv_path, columns_to_anonymize=(), flows_per_file=0, rotate_files=0)
                except ValueError:
           
                    print(not os.path.exists(self.csv_path))
                    continue

                else:
                    print('pass')
                    break
```

**Comment by drnpkr** [comment]
Can you share the PCAP with us, please?

**Reply by DarmJarlo** [reply]
Hey thank you for your reply. I found the problem is actually that the pcap file cannot be created in time. And so I am trying to modify the source code of NFstream to let it directly convert a list of collected packets ( collected by the application I wrote on the ryu controller) instead of converting a  pcap file or packets from a monitored interface. So I met some problems again, as https://github.com/nfstream/nfstream/discussions/167     Could you please help me here? Thanks very much!
